sequencing-server/src/lib/codegen/CommandEDSLPreface.ts

```diff
@@ -7,6 +7,15 @@ export enum TimingTypes {
   COMMAND_COMPLETE = 'COMMAND_COMPLETE',
 }
 
+export type VariableOptions = {
+  name: string;
+  type: 'FLOAT' | 'INT' | 'STRING' | 'UINT' | 'ENUM';
+  enum_name?: string | undefined;
+  allowable_values?: unknown[] | undefined;
+  // @ts-ignore : 'VariableRange' found in JSON Spec
+  allowable_ranges?: VariableRange[] | undefined;
+};
+
```

Comment:
I think you could add an extra type here that would require the `enum_name` field if the type was set to `ENUM`, something like:
```
type DefaultVariableOptions = {
  name: string;
  type: 'FLOAT' | 'INT' | 'STRING' | 'UINT';
  allowable_values?: unknown[] | undefined;
  allowable_ranges?: number[] | undefined;
}

type EnumVariableOptions = {
  name: string;
  type: 'ENUM'
  enum_name: string;
};

type VariableOptions = DefaultVariableOptions | EnumVariableOptions;

const testVar: VariableOptions = { // Type '{ name: string; type: "ENUM"; }' is not assignable to type 'VariableOptions'. Property 'enum_name' is missing in type '{ name: string; type: "ENUM"; }' but required in type 'EnumVariableOptions'.
  name: 'test',
  type: 'ENUM'
};
```

Reply:
It starts to get tricky as I am not sure how to limit the available functions. For example, If I provide a`DefaultVariableOptions ` I still have access to the `ENUM_NAME` function to set the enum name thus getting us to the error of defining an enum name without the enum type. Ideally, I would not allow the user to have access to that function.

`VARIABLE('duration', 'FLOAT').ENUM_NAME("DURATION_TYPE")`

Reply:
Ah, yeah that makes sense. I also wrote this comment before I scrolled down and saw the `Variable` class.

Reply:
No worries, It was a good idea.